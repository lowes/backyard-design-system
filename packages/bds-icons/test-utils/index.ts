import { customRender } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }
