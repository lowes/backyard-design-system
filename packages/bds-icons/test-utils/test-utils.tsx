import * as React from 'react'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import * as theme from '@lowes-tech/bds-tokens/v2/light/theme'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
const customRender = (ui, options) => render(ui, { wrapper: RenderWrapper, ...options })

export { customRender }
