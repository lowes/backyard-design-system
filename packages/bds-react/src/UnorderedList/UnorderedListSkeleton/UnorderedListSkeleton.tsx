import * as React from 'react'
import classnames from 'classnames'
import type { BDSForwardRef } from '../../utils/typings/BDSComponentProps'

import Skeleton from '../../Skeleton'
import SkeletonWrapper from './styles'

/**
 * Backard UnorderedListSkeleton component.
 * 
 * Returns a skeleton component with a default width of 300px and a default
 * quantity of list items set to 4. The density is also set to normal by 
 * default.
 * 
 * <UnorderedListSkeleton />
*/

const UnorderedListSkeleton: BDSForwardRef<UnorderedListSkeletonProps> = React.forwardRef<
    UnorderedListSkeletonRef,
    UnorderedListSkeletonProps
>(function UnorderedListSkeleton (
    {
        animate = false,
        items = 4,
        density = 'normal',
        width = '18.75rem',
        className,
        ...props
    },
    ref,
) {

    const rects = [...new Array(items)]

    return (
        <SkeletonWrapper 
            className={classnames(className, 'Unordered-list--skeleton', `density--${density}`)} 
            width={width}
            {...props}
            ref={ref}
        >
            {
                rects.map((item, index) => (
                    <Skeleton 
                        key={`skeleton-ol-${index}`}
                        variant='rect'
                        height='size_24'
                        width='100%'
                        animate={animate}
                        className='list-item--skeleton'
                    />
                ))
            }
        </SkeletonWrapper>
    )
})

type UnorderedListSkeletonRef = HTMLDivElement

interface UnorderedListSkeletonProps {
    animate?: boolean
    density?: string
    items?: number
    width?: string
    className?: string
}

UnorderedListSkeleton.bdsName = 'UnorderedListSkeleton'

export { UnorderedListSkeleton }

export type { UnorderedListSkeletonProps, UnorderedListSkeletonRef }

export default UnorderedListSkeleton