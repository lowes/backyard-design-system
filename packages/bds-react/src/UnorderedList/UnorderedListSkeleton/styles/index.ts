import styled from 'styled-components'
import type { UnorderedListSkeletonProps } from '../UnorderedListSkeleton'

const SkeletonWrapper = styled.div<UnorderedListSkeletonProps>`
    display: flex;
    flex-direction: column;
    width: ${({ width }) => width};

    &.density--normal {
        .variant--rect {
            margin-top: ${({ theme }) => theme.sizes.size_8};
        }
    }

    &.density--comfort {
        .variant--rect {
            margin-top: ${({ theme }) => theme.sizes.size_12};
        }
    }

    &.density--condensed {
        .variant--rect {
            margin-top: ${({ theme }) => theme.sizes.size_4};
        }
    }
`

export { SkeletonWrapper }
export default SkeletonWrapper