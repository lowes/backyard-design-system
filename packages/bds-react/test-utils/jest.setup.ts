import * as React from 'react'
import '@testing-library/jest-dom'
import 'jest-styled-components'
import Enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

Enzyme.configure({ adapter: new Adapter() })

jest.setTimeout(1000 * 60 * 20)

// Mock `useRandomId` hook
// gives consistent ids instead
// of randomly generated ones for
// snapshot tests
let mockId = 0

jest.mock('../src/utils/hooks/useRandomId', () => {
    return jest.fn((id) => id || `${mockId++}`)
})

// Before each test...
beforeEach(() => {
    // Reset `useRandomId` counter
    mockId = 0
})
