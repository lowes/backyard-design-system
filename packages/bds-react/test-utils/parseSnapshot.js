const parseSnapshot = (content) => {
    const data = get_data(content)

    return Object.keys(data).reduce(
        (parsed, key) => ({ ...parsed, [key]: parse(data[key]) }),
        {},
    )
}

function parse(value) {
    const reverted_value = revert_extra_line_breaks(value).replace(/\n/g, '\\n')

    try {
        return reverted_value === 'undefined'
            ? undefined
            : JSON.parse(reverted_value)
    } catch (error) {
        // istanbul ignore next
        throw new Error(
            `Unexpected error while parsing:\n\n>>> start >>>\n${reverted_value}\n<<< end <<<\n\n${
            error.message
            }`,
        )
    }
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L128
function revert_extra_line_breaks(str) {
    return str.includes('\n') ? str.slice(1, -1) : str
}

// https://github.com/facebook/jest/blob/v20.0.4/packages/jest-snapshot/src/utils.js#L98
function get_data(content) {
    const data = Object.create(null)
    const populate = new Function('exports', content)
    
    populate(data)
    
    return data
}

export {
    parseSnapshot
}

export default parseSnapshot
