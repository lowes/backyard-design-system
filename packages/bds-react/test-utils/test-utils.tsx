import * as React from 'react'
import { render } from '@testing-library/react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/ThemeProvider'

/**
 * Render Wrapper Helper
 *
 * This component wraps all test renders from react-testing-library
 * Uses default ThemeProvider with light
 */
const RenderWrapper = ({ children }) => {
    return <ThemeProvider theme="light">{children}</ThemeProvider>
}

/**
 * Function to override custom render with
 *
 * @param {*} ui
 * @param {*} options
 */
// @ts-ignore
const customRender = (ui, options = {}) => render(ui, { wrapper: RenderWrapper, ...options })

/**
 * Function to override custom mount with
 *
 * @param {*} ui
 * @param {*} options
 */
const customMount = (ui, options = {}) =>
    mount(ui, {
        wrappingComponent: ThemeProvider,
        wrappingComponentProps: { theme: 'light' },
        ...options,
    })

export { customRender, customMount }
