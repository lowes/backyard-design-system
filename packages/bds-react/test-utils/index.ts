// @ts-ignore
import { customRender, customMount } from './test-utils'

// Export all from react-testing-library
export * from '@testing-library/react'

// Export Simulate from React-DOM if needed
export { Simulate, act } from 'react-dom/test-utils'

// Override render method
export { customRender as render }

// Override shallow method
export { customMount as mount }
