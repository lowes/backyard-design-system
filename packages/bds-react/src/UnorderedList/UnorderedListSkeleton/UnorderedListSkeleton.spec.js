import * as React from 'react'
import { render } from '../../../test-utils'
import UnorderedListSkeleton from './UnorderedListSkeleton'

describe('UnorderedListSkeleton Snapshots', () => {
    test('UnorderedListSkeleton default snapshot', () => {
        const { asFragment } = render(
            <UnorderedListSkeleton />
        )

        expect(asFragment()).toMatchSnapshot()
    })


    test('UnorderedListSkeleton animate snapshot', () => {
        const { asFragment } = render(
            <UnorderedListSkeleton animate={true} />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('UnorderedListSkeleton multiple options snapshot', () => {
        const { asFragment } = render(
            <UnorderedListSkeleton items={5} />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('UnorderedListSkeleton custom width snapshot', () => {
        const { asFragment } = render(
            <UnorderedListSkeleton width='30rem' />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('UnorderedListSkeleton comfort density snapshot', () => {
        const { asFragment } = render(
            <UnorderedListSkeleton density='comfort' />
        )

        expect(asFragment()).toMatchSnapshot()
    })

    test('UnorderedListSkeleton condensed density snapshot', () => {
        const { asFragment } = render(
            <UnorderedListSkeleton density='condensed' />
        )

        expect(asFragment()).toMatchSnapshot()
    })
})