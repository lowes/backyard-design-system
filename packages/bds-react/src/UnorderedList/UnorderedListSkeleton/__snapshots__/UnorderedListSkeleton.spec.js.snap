// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnorderedListSkeleton Snapshots UnorderedListSkeleton animate snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 100%;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 100%;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 18.75rem;
}

.c0.density--normal .variant--rect {
  margin-top: 0.5rem;
}

.c0.density--comfort .variant--rect {
  margin-top: 0.75rem;
}

.c0.density--condensed .variant--rect {
  margin-top: 0.25rem;
}

<div
    class="c0 Unordered-list--skeleton density--normal"
    width="18.75rem"
  >
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--animated"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--animated"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--animated"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--animated"
      height="size_24"
      width="100%"
    />
  </div>
</DocumentFragment>
`;

exports[`UnorderedListSkeleton Snapshots UnorderedListSkeleton comfort density snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 100%;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 100%;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 18.75rem;
}

.c0.density--normal .variant--rect {
  margin-top: 0.5rem;
}

.c0.density--comfort .variant--rect {
  margin-top: 0.75rem;
}

.c0.density--condensed .variant--rect {
  margin-top: 0.25rem;
}

<div
    class="c0 Unordered-list--skeleton density--comfort"
    width="18.75rem"
  >
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
  </div>
</DocumentFragment>
`;

exports[`UnorderedListSkeleton Snapshots UnorderedListSkeleton condensed density snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 100%;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 100%;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 18.75rem;
}

.c0.density--normal .variant--rect {
  margin-top: 0.5rem;
}

.c0.density--comfort .variant--rect {
  margin-top: 0.75rem;
}

.c0.density--condensed .variant--rect {
  margin-top: 0.25rem;
}

<div
    class="c0 Unordered-list--skeleton density--condensed"
    width="18.75rem"
  >
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
  </div>
</DocumentFragment>
`;

exports[`UnorderedListSkeleton Snapshots UnorderedListSkeleton custom width snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 100%;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 100%;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 30rem;
}

.c0.density--normal .variant--rect {
  margin-top: 0.5rem;
}

.c0.density--comfort .variant--rect {
  margin-top: 0.75rem;
}

.c0.density--condensed .variant--rect {
  margin-top: 0.25rem;
}

<div
    class="c0 Unordered-list--skeleton density--normal"
    width="30rem"
  >
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
  </div>
</DocumentFragment>
`;

exports[`UnorderedListSkeleton Snapshots UnorderedListSkeleton default snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 100%;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 100%;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 18.75rem;
}

.c0.density--normal .variant--rect {
  margin-top: 0.5rem;
}

.c0.density--comfort .variant--rect {
  margin-top: 0.75rem;
}

.c0.density--condensed .variant--rect {
  margin-top: 0.25rem;
}

<div
    class="c0 Unordered-list--skeleton density--normal"
    width="18.75rem"
  >
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
  </div>
</DocumentFragment>
`;

exports[`UnorderedListSkeleton Snapshots UnorderedListSkeleton multiple options snapshot 1`] = `
<DocumentFragment>
  .c1 {
  background: var(--bds-color-skeleton-01);
  position: relative;
  padding: 0;
  border: none;
  box-shadow: none;
  pointer-events: none;
  overflow: hidden;
}

.c1.style--animated::before {
  position: absolute;
  content: '';
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background: var(--bds-color-skeleton-01);
  -webkit-animation-name: pulse;
  animation-name: pulse;
  -webkit-animation-duration: 2s;
  animation-duration: 2s;
  -webkit-animation-timing-function: ease-in-out;
  animation-timing-function: ease-in-out;
  -webkit-animation-delay: 0s;
  animation-delay: 0s;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
  -webkit-animation-direction: normal;
  animation-direction: normal;
  -webkit-animation-fill-mode: none;
  animation-fill-mode: none;
  -webkit-animation-play-state: running;
  animation-play-state: running;
}

.c1.variant--circle {
  width: 100%;
  height: 1.5rem;
  border-radius: var(--bds-border-radius-circle);
}

.c1.variant--rect {
  width: 100%;
  height: 1.5rem;
}

.c1.variant--rect.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--rect.shape--squared {
  border-radius: 0;
}

.c1.variant--text {
  width: 100%;
  height: 16px;
  margin-bottom: var(--bds-sizes-size-6);
}

.c1.variant--text.shape--rounded {
  border-radius: var(--bds-border-radius-lg);
}

.c1.variant--text.shape--squared {
  border-radius: 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 18.75rem;
}

.c0.density--normal .variant--rect {
  margin-top: 0.5rem;
}

.c0.density--comfort .variant--rect {
  margin-top: 0.75rem;
}

.c0.density--condensed .variant--rect {
  margin-top: 0.25rem;
}

<div
    class="c0 Unordered-list--skeleton density--normal"
    width="18.75rem"
  >
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
    <div
      class="c1 variant--rect shape--rounded list-item--skeleton style--default"
      height="size_24"
      width="100%"
    />
  </div>
</DocumentFragment>
`;
